core/kazoo_numbers/src/knm_phone_number.erl

```diff
@@ -74,7 +74,7 @@
 %% Without 'undefined' there, Dialyzer outputs 'false' positives.
 %% It has trouble inferring what is happening in from_json/1's setters.
 %% And all this is because we need to set is_dirty reliably.
--record(knm_phone_number, {number :: kz_term:api_ne_binary()             %%%
+-record(knm_phone_number, {number :: kz_term:ne_binary()                 %%%
```

Comment:
since you know this change makes dialyzer unhappy with the empty `#knm_phone_number{}` record creation, why make it?

Reply:
It also makes Dialyzer not happy in set_number/2 with the `{undefined, undefined}` clause

Reply:
I didn't add that comment, but anyway when  I removed it dialyzer didn't report anything. How did you ran dialyzer to see it is failing?

Reply:
I removed it because I thought it will fix the issue with `knm_errors:by_carrier` dialyzer warning in `knm_telnyx_util:rep/1`, but when after I fix that by_carrier, dialyzer didn't report any issue with that record field.